### PR TITLE
FM2-563: Medication Dispense: Add Recorded Extension

### PR DESCRIPTION
**Pull request by mogoodrich**
## Issue I worked on

see https://issues.openmrs.org/browse/FM2-563

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Review by mseaton — APPROVED** [review]
Code looks fine.  Defer to @ibacher

File: api/src/main/java/org/openmrs/module/fhir2/FhirConstants.java [hunk 1]
```diff
@@ -112,6 +112,8 @@ private FhirConstants() {
 	public static final String OPENMRS_FHIR_EXT_OBSERVATION_REFERENCE_RANGE = OPENMRS_FHIR_EXT_PREFIX
 	        + "/obs/reference-range";
 	
+	public static final String OPENMRS_FHIR_EXT_RECORDED = OPENMRS_FHIR_EXT_PREFIX + "/recorded";
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/FhirConstants.java:115]:
```suggestion
	public static final String OPENMRS_FHIR_EXT_RECORDED = OPENMRS_FHIR_EXT_PREFIX + "/dispense/recorded";
```

**Review by ibacher — APPROVED** [review]

**Comment by ibacher** [comment]
I changed the URI for the extension to keep it semi-namespaced, but otherwise, LGTM.

**Comment by mogoodrich** [comment]
Thanks @ibacher , I will merge!

**Comment by ibacher** [comment]
> Patch coverage: 100.00%

I don't think I've ever seen 100% patch coverage before...

**Pull request merged**